### PR TITLE
added implementation for loss function

### PR DESCRIPTION
**Pull request by vipulpandey12345**
updated loss implementation -> this function relies on getting labels from the data generator

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -8,15 +8,30 @@
...
 
 
-
 class CustomLossFunction(nn.Module):
-    def __init__(self):
+    def __init__(self, margin=1.0):
         super(CustomLossFunction, self).__init__()
-        # Initialize your loss components here, if any
+        self.margin = margin
+        self.model = SentenceTransformer('distilbert-base-uncased')
```
Comment (Sharwin24) [inline train.py]: I don't know if this parameter is ever something else but this string should probably be passed into the `init` function as a parameter with a default value

File: train.py [hunk 2]
```diff
@@ -8,15 +8,30 @@
...
+        self.training_examples = gptdatagenerator.generate_qa()  # Assuming gptdatagenerator generates input examples
```
Comment (Sharwin24) [inline train.py]: We might want to create a function here or in `gptdatagenerator` to determine if it has generated examples or not, log appropriately, and then either continue or not if the obtained examples exist or not

**Review by Sharwin24 — COMMENTED** [review]
Just reading over, looks like the loss function depends on `GPTDataGenerator` so we should add some logging or checks to make sure that module has done whatever it needs to before running this function
